lib/SILOptimizer/Transforms/AllocBoxToStack.cpp

```diff
@@ -872,19 +978,25 @@ static void rewritePartialApplies(AllocBoxToStackState &pass) {
   // partial_apply.
   SILOptFunctionBuilder FuncBuilder(*pass.T);
   for (auto &It : IndexMap) {
-    auto *PartialApply = It.first;
+    auto &Apply = It.first;
     auto &Indices = It.second;
 
     // Sort the indices and unique them.
     std::sort(Indices.begin(), Indices.end());
     Indices.erase(std::unique(Indices.begin(), Indices.end()), Indices.end());
 
-    PartialApplyInst *Replacement =
-      specializePartialApply(FuncBuilder, PartialApply, Indices, pass);
-    PartialApply->replaceAllUsesWith(Replacement);
-
-    auto *FRI = cast<FunctionRefInst>(PartialApply->getCallee());
-    PartialApply->eraseFromParent();
+    if (Apply.getKind() == ApplySiteKind::PartialApplyInst) {
```

Comment:
Will this work for all applies including try_apply, begin_apply? My thought: Add an exhaustive switch with an assert for the other cases. This will also make sure people update this code if we add additional apply sites.

Reply:
Added support for try_apply and begin_apply now.